tests/integ/test_function.py

```diff
@@ -206,12 +206,22 @@ def test_order(session):
     ]
 
 
+@pytest.mark.localtest
 def test_current_date_and_time(session):
-    df1 = session.sql("select current_date(), current_time(), current_timestamp()")
-    df2 = session.create_dataframe([1]).select(
+    max_delta = 1
+    df = session.create_dataframe([1]).select(
         current_date(), current_time(), current_timestamp()
     )
-    assert len(df1.union(df2).collect()) == 1
+    rows = df.collect()
+
+    assert len(rows) == 1, "df1 should only contain 1 row"
+    date, time, timestamp = rows[0]
+    time1 = datetime.datetime.combine(date, time).timestamp()
+    time2 = timestamp.timestamp()
+
+    assert time1 == pytest.approx(
+        time2, max_delta
+    ), f"Times should be within {max_delta} seconds of each other."
```

Comment:
why do we have this test here? I'm wondering if this would introduce test flakiness in live mode.

Reply:
This seemed like the easiest way to get two times that should be close to each other. Testing an exact value would be ideal, but since we don't want to mock out the server response the next best thing would be to have a fuzzy equality. 

If this causes flakiness we can relax the allowed delta. If we relaxed the delta to our server response SLA a failure would not be due to flakiness, but due to a larger issue in my opinion.

Reply:
in this case, should we also test `timestamp` with `time2`?

Reply:
gotcha, but I'm still wondering whether it's needed to compare time1 with time2..

previously the test is comparing the results from raw sql execution with from df.select function -- the same results expected to be the same because after union, the new df is supposed to only have one row.

we can keep the test here, it does no harm